Dockerfile

```diff
@@ -1,21 +1,27 @@
-FROM alpine AS common
-RUN echo "policy-server:x:65533:65533::/tmp:/sbin/nologin" >> /etc/passwd
-RUN echo "policy-server:x:65533:policy-server" >> /etc/group
+FROM rust:1.70 AS build
+WORKDIR /usr/src
+
+# Download the target for static linking.
+RUN rustup target add $(arch)-unknown-linux-musl
 
-# amd64-specific
-FROM scratch AS build-amd64
-COPY --from=common /etc/passwd /etc/passwd
-COPY --from=common /etc/group /etc/group
-COPY --chmod=0755 policy-server-x86_64 /policy-server
+# Fix ring building using musl - see https://github.com/briansmith/ring/issues/1414#issuecomment-1055177218
+RUN apt-get update && apt-get install musl-tools clang llvm -y
+ENV CC="clang"
```

Comment:
I suppose this is the only way of setting the compiler, and we can't do it on the cargo.toml file.
It would be nice to have the same way of building a local release binary, and the container image (apart from having DRY).

Reply:
Unfortunately this cannot be set into the `.cargo/config.toml`. I really tried hard, but I couldn't find a way to do that.

The local build (`cargo build`) is going to use gcc and is not going to use musl. Do we want to change that to use musl too? I think this should be possible, but I'm not entirely sure about that.

However, keep in mind we will introduce the following prerequisites:

* Have clang installed
* Have the musl target added
* Have something like [direnv](https://direnv.net/) installed, so that we can ensure the `CC` is set accordingly

Do you want me to give it a try?

Reply:
I'm ok as it is, resolving.